### PR TITLE
prov/efa: fix 0 byte write and add unit test

### PR DESCRIPTION
**Pull request by wzamazon**
fix efa_rdm_ope_post_remote_write for 0 byte, and add a unit test.

### CODE REVIEW

File: prov/efa/src/rdm/efa_rdm_ope.c [hunk 1]
```diff
@@ -1459,10 +1459,34 @@ int efa_rdm_ope_post_remote_write(struct efa_rdm_ope *ope)
...
-
 	efa_rdm_ope_try_fill_desc(ope, 0, FI_WRITE);
 	ep = ope->ep;
+	if (ope->bytes_write_total_len == 0) {
+		/* According to libfabric document
+		 *     https://ofiwg.github.io/libfabric/main/man/fi_rma.3.html
+		 * write with 0 byte is allowed.
+		 *
+		 * Note that because send operation used a pkt_entry as wr_id,
+		 * we had to use a pkt_entry as context for read too.
```
Comment (shijin-aws) [inline prov/efa/src/rdm/efa_rdm_ope.c:1470]: read or write?

File: prov/efa/src/rdm/efa_rdm_ope.c [hunk 2]
```diff
@@ -1459,10 +1459,34 @@ int efa_rdm_ope_post_remote_write(struct efa_rdm_ope *ope)
...
+					0,
+					ope->desc[0],
+					ope->rma_iov[0].addr,
+					ope->rma_iov[0].key);
+		if (err)
+			efa_rdm_pke_release_tx(ep, pkt_entry);
+		return err;
+	}
+
+	assert(ope->bytes_write_submitted < ope->bytes_write_total_len);
```
Comment (shijin-aws) [inline prov/efa/src/rdm/efa_rdm_ope.c:1489]: could you remind me again why code before your change cannot handle 0 byte?

File: prov/efa/src/rdm/efa_rdm_ope.c (hunk 1 above)
Comment (wzamazon) [reply prov/efa/src/rdm/efa_rdm_ope.c]: should be write. fixed.

File: prov/efa/src/rdm/efa_rdm_ope.c (hunk 2 above)
Comment (wzamazon) [reply prov/efa/src/rdm/efa_rdm_ope.c:1489]: So line 1510, we used a loop to post write, until `bytes_write_submitted` equal `bytes_write_total_len`.

But for 0 bytes write, `bytes_write_total_len` is 0, so the loop won't start.

Comment (shijin-aws) [reply prov/efa/src/rdm/efa_rdm_ope.c:1489]: Ah, OK, thanks!

**Review by shijin-aws — APPROVED** [review]

**Comment by shijin-aws** [comment]
need a backport

**Pull request merged**